spring-projectsGH-3169: Fix DefaultSessionFactoryLocator addSessionFactory key's type from String to Objec

kezhevatov opened the pull request:
spring-projectsGH-3169: Fix DefaultSessionFactoryLocator addSessionFactory key's type from String to Object

All other DefaultSessionFactoryLocator contracts are based on the Object, so this addSessionFactory has to be on Object as well.

artembilan commented on spring-integration-file/src/main/java/org/springframework/integration/file/remote/session/DefaultSessionFactoryLocator.java:
```diff
@@ -54,7 +54,7 @@ public DefaultSessionFactoryLocator(Map<Object, SessionFactory<F>> factories, Se
 	 * @param factory the factory.
 	 */
-	public void addSessionFactory(String key, SessionFactory<F> factory) {
+	public void addSessionFactory(Object key, SessionFactory<F> factory) {
```
How about this my concern: https://github.com/spring-projects/spring-integration/issues/3169#issuecomment-583058147 ?

Whenever we are going to back-port it we should keep backward compatibility, so an upgrade for the target project will be smooth. Even if we don't back-port it into a previous version, we still need to keep binary compatibility for current `5.3` version: Spring Cloud is going to support both Spring Boot 2.2 and 2.3 generations.

Therefore my concern about deprecating the current method and introduce a new one like you have done.

Thanks

kezhevatov replied:
Done. Thanks for quick answer

artembilan commented on spring-integration-file/src/main/java/org/springframework/integration/file/remote/session/DefaultSessionFactoryLocator.java:
```diff
@@ -53,11 +53,23 @@ public DefaultSessionFactoryLocator(Map<Object, SessionFactory<F>> factories, Se
 	 */
+	@Deprecated
 	public void addSessionFactory(String key, SessionFactory<F> factory) {
 		this.factories.put(key, factory);
```
I have an opinion that deprecated API should call a new one for better coverage.
This way when we need to make a change in the target logic, a deprecated API won't be affected because it is meant to be for removal eventually anyway. 😄

kezhevatov replied:
Yep, it make sense. Done

kezhevatov committed:
spring-projectsGH-3169: Fix DefaultSessionFactoryLocator addSessionFactory key's type from String to Object

All other DefaultSessionFactoryLocator contracts are based on the Object, so this addSessionFactory has to be on Object as well.

artembilan reviewed (approved):
Merging...

I'll add your name to the class author list and do some clean up locally...

artembilan commented:
Merged as https://github.com/spring-projects/spring-integration/commit/edf84a393d4dd8f5db8de0d0fdba581be86efa50 and cherry-picked to `5.2.x`.

@kezhevatov ,

thank you for the contribution; looking forward for more!

For the future, please, consider to test the project against your changes: you have missed to fix a deprecated method usage in the `DelegatingSessionFactoryTests`.

Plus it would be great to make a good commit message (not a PR description, but commit): https://chris.beams.io/posts/git-commit/
